src-go/server/internal/tls/factory.go

```diff
@@ -31,6 +33,9 @@ func (f *FactoryWithClientHelloId) NewUTLSConn(conn net.Conn, config *tls.Config
 	if clientHelloID == nil {
 		clientHelloID = DefaultClientHelloID
 	}
+	if f.ClientHelloSpec != nil {
+		clientHelloID = &utls.HelloCustom
+	}
```

Comment:
The `ClientHelloSpec` field has priority over `clientHelloID`, so there's no need to check for `clientHelloID` if `ClientHelloSpec` is already set:

```suggestion
        if f.ClientHelloSpec != nil {
		clientHelloID = &utls.HelloCustom
	} else if clientHelloID == nil {
		clientHelloID = DefaultClientHelloID
	}
	
```